Feat: Style Hero PLP

eduardoformiga opened the pull request:
## What's the purpose of this pull request?
This PR aims to adapt the `Hero` component so that we can use it on the `plp` page as well.
Now we can pass new props: 
- `icon`
- `variants`: `small` and `default`.

## Preview
|Homepage (variant = default)|PLP (variant = small)|
|-|-|
|![homehero](https://user-images.githubusercontent.com/11325562/149254546-d805d065-3164-4e29-82dc-726c40747f05.gif)|![plp](https://user-images.githubusercontent.com/11325562/149254253-68c43434-47ce-4ec3-9d27-aa1bb4068307.gif)|
## How to test it
👀   Preview: https://preview-230--base.preview.vtex.app

## References
[FSSS-114](https://vtex-dev.atlassian.net/jira/software/projects/FSSS/boards/632?selectedIssue=FSSS-114)

eduardoformiga committed:
Creates Hero small variant and adapt last implementation

eduardoformiga committed:
Adds Hero on plp

eduardoformiga committed:
Tweaks style and breakpoints

hellofanny commented on src/components/ui/Hero/hero.scss:
```diff
@@ -63,23 +48,69 @@
+      @include media(">=notebook") { padding-left: var(--space-7); }
+    }
+
+    .title-hero {
```
Niceeee!

hellofanny commented on src/components/ui/Hero/hero.scss:
```diff
@@ -63,23 +48,69 @@
+    p {
+      font-size: var(--space-3);
+
+      @include media(">=tablet") { font-size: 1.25rem; }
```
```suggestion
      @include media(">=tablet") { font-size: var(--text-size-3); }
```

eduardoformiga replied:
ohhh my, completely forgot these typography variables, hahaha.

hellofanny commented on src/components/ui/Hero/hero.scss:
```diff
@@ -63,23 +48,69 @@
+    }
+
+    p {
+      font-size: var(--space-3);
```
```suggestion
      font-size: var(--text-size-2);
```

hellofanny commented on src/components/ui/Hero/hero.scss:
```diff
@@ -63,23 +48,69 @@
+    .title-hero {
+      font-size: var(--space-5);
+
+      @include media(">=tablet") { font-size: var(--space-7); }
```
```suggestion
      @include media(">=tablet") { font-size: var(--text-size-7); }
```

hellofanny commented on src/components/ui/Hero/hero.scss:
```diff
@@ -63,23 +48,69 @@
+    }
+
+    .title-hero {
+      font-size: var(--space-5);
```
```suggestion
      font-size: var(--text-size-5);
```

lucasfp13 reviewed (approved):
After review and apply the code suggestions, LGTM!

hellofanny reviewed (changes requested):
Some adjusts and LGTM! 🚀 🚀 🚀

eduardoformiga committed:
Update src/components/ui/Hero/hero.scss

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

eduardoformiga committed:
Update src/components/ui/Hero/hero.scss

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

eduardoformiga committed:
Update src/components/ui/Hero/hero.scss

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

eduardoformiga committed:
Update src/components/ui/Hero/hero.scss

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

eduardoformiga commented:
> 

Completely forgot these typography variables! Good catch! 😄

hellofanny commented:
I also noticed something else: 🤔 

|Before|
|-|
![image](https://user-images.githubusercontent.com/3356699/149237541-b6a76973-3503-4ae9-a6b2-6b1ebb9af0cd.png)

|After|
|-|
![image](https://user-images.githubusercontent.com/3356699/149237573-12ca7496-bcce-491d-be1e-020624a4f96c.png)

eduardoformiga committed:
Fixes Hero default style

eduardoformiga committed:
Tweaks Hero style

eduardoformiga commented:
> I also noticed something else: 🤔 Can you please check this
> 
> Browser: Chrome Size: 1920x1080
> [...] 


@hellofanny. Thank you for the heads up!
I changed the implementation now focusing on keeping the margin and padding following the hole page as a reference.
I also updated the images in the description considering both variants.

renatamottam commented on src/styles/typography.scss:
```diff
@@ -28,14 +28,6 @@ h1, h2, h3, h4, h5, h6, p, ul {
 
 // Titles ///////////////////////////
 
-.title-hero {
-  font-weight: var(--text-weight-black);
-  font-size: var(--space-5);
-  line-height: 1.1;
-
-  @include media(">=notebook") { font-size: var(--space-8); }
-}
-
```
Don't know if I like removing it from here, it's useful for future customization to see all the major typography sizes on the same page.

hellofanny replied:
Hm! That is something to consider.
So, just let's keep it here? 👀

eduardoformiga replied:
We decided together to keep this on typography and adapt in hero. 🎉

renatamottam commented on src/components/ui/Hero/hero.scss:
```diff
@@ -6,7 +6,7 @@
   background-color: var(--bg-secondary-default);
 
-  @include media(">=notebook") {
+  @include media(">=tablet") {
```
I have an idea to fix this, slack me!
<img src="https://user-images.githubusercontent.com/11613011/149336852-6dcc5bf3-62a4-4991-98e7-b34c2537dc41.png" width="400" />

hellofanny reviewed (approved):
Well done!! 🚀 🚀 🚀 After consider what renata said I think we are ready to go!

eduardoformiga committed:
Tweaks title-hero and paragraph style

eduardoformiga committed:
Tweaks image and its internal wrapper styles

filipewl commented:
@renatamottam is going to make adjustments later and [said](https://vtex.slack.com/archives/C02MVS48Y0Z/p1642164807328100?thread_ts=1642162598.324800&cid=C02MVS48Y0Z) we could merge this!
